src/drive/web/modules/navigation/FileExplorer.jsx

```diff
@@ -187,32 +190,12 @@ export const handleFileOpen = (file, availableOffline, props, dispatch) => {
   }
 
   const isNote = models.file.isNote(file)
-  //This is a tmp hack since Notes doesn't handle sharing properly.
-  const isMyNote = isNote && isNoteMine(file, props.client)
-
-  if (isMyNote) {
-    const notesInstalled = models.applications.isInstalled(
-      props.apps.data,
-      notesApp
-    )
-    let url = ''
-    if (notesInstalled) {
-      url = models.note.generateUrlForNote(
-        models.applications.getUrl(notesInstalled),
-        file
-      )
-    } else {
-      url = models.applications.getStoreInstallationURL(
-        props.apps.data,
-        notesApp
-      )
-    }
-    if (url !== '') {
-      window.location.href = url
-    } else {
-      //If something went wrong previoulsy like the request to /apps has failled
-      //we display the note within the viewer
-      redirectToViewer()
+  const { client } = props
+  if (isNote) {
+    try {
+      window.location.href = await fetchUrlToOpenANote(client, file)
+    } catch (e) {
+      Alerter.error('alert.offline')
```

Comment:
We don't want to redirect to the viewer in this case?

Reply:
Not yet, next PR. I've to update the Viewer to display a few lines on it. And for that I need to upgrade UI -> XP modal and so on :p